application-collabora-ui/src/main/resources/Collabora/Code/Main.xml

```diff
@@ -219,6 +266,26 @@
     }).fail(function(jqxhr, textStatus, error) {
       new XWiki.widgets.Notification(errorMessage + error, 'error');
     });
+
+    // Listen to messages send by Collabora server.
+    window.addEventListener("message", receiveMessage, false);
+  });
+
+  /**
+   * Notify users of unsaved changes when they leave the editor.
+   */
+  const menuActions = '.actionMenu .homePage, .actionMenu .fileHomePage, .actionMenu .closeEditor';
+  $(document).on('click', menuActions, function(e) {
```

Comment:
What if the user closes the browser tab / window. In CKEditor integration I've used instead the ``beforeunload`` event, see https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/resources/xwiki-save/plugin.js#L95-L100 . I don't know if this could work you also.

Reply:
It looks like I cannot use it. The `beforeunload` event is not always working, and when it works, even if I select `stay on page`, the Collabora editor is still unloaded and a `View_Removed` message is send by the iframe.

Reply:
OK, let's keep it like this for now.